code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,182 @@
+///////////////////////////
+// Dost thou even hoist? //
+///////////////////////////
+
+/obj/item/hoist_kit
+	name = "Hoist Kit"
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	icon = 'icons/obj/forensics.dmi'
+	icon_state = "case"
+
+/obj/item/hoist_kit/attack_self(mob/user)
+	new /obj/structure/hoist (get_turf(user), user.dir)
+	user.visible_message(span("warning", "[user] deploys the hoist kit!"), span("notice", "You deploy the hoist kit!"), span("notice", "You hear the sound of parts snapping into place."))
+	qdel(src)
+
+/obj/effect/hoist_hook
+	name = "Hoist Clamp"
+	desc = "A clamp used to lift people or things."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "up"
+	var/obj/structure/hoist/source_hoist
+	can_buckle = 1
+	anchored = 1
+
+/obj/effect/hoist_hook/MouseDrop_T(atom/movable/M,mob/user)
+	if(!istype(M, /atom/movable/)) // why this is even necessary, i don't know.
+		return
+	if (initial(anchored))
+		return
+	if (istype(M, /obj/structure/))
+		return
+	if (source_hoist.hoistee)
+		return
+	source_hoist.hoistee = M
+	if (!Adjacent(M))
+		return
+	if (get_turf(M) != get_turf(src))
+		M.Move(get_turf(src))
+	if (istype(M, /mob/living/))
+		user_buckle_mob(M, user)
+	else
+		M.anchored = 1
+		user.visible_message(span("danger", "[user] attaches \the [M] to the hoist clamp."), span("danger", "You attach \the [M] to the hoist clamp."), span("danger", "You hear something clamp into place."))
+
+/obj/effect/hoist_hook/MouseDrop(atom/dest)
+	if(!usr || !dest) return
+	if(!Adjacent(usr) || !dest.Adjacent(usr)) return // carried over from the default proc
+
+	spawn(0)
```

Comment:
I just copied it directly from the default `MouseDrop` proc. There's probably a better way to go about it, yes.